### PR TITLE
[BEAM-13904] Increase unit testing in the reflectx package

### PR DESCRIPTION
**Pull request by damccorm**
Adds testing for everything in reflectx except for calls.go (generated code) and tags.go (unused as best I can tell, PR to deprecate coming shortly).

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Review by jrmccluskey — APPROVED** [review]
LGTM

**Comment by damccorm** [comment]
> Bot seems to have hit a snag of some kind.

Yeah, we just need to give it the right permissions set - I just put up https://github.com/apache/beam/pull/17031 which should fix it

**Comment by damccorm** [comment]
FWIW, the bot was trying to ping Daniel - https://github.com/apache/beam/runs/5453133956?check_suite_focus=true - it just ran into auth issues because the token GitHub gave us scopes down on PRs by default (its a security decision which does make sense, I just hadn't been able to test that interaction well because the Beam's security settings are mostly opaque to me at the moment).

**Review by youngoli — APPROVED** [review]
Looks good to me, just a few nits before merging in.

File: sdks/go/pkg/beam/core/util/reflectx/call_test.go [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+func TestCallNoPanic(t *testing.T) {
+	RegisterFunc(reflect.TypeOf((*func(string) string)(nil)).Elem(), mapperStringMaker)
+	fn := func(str string) string {
+		return string(str)
+	}
+	madeFn := MakeFunc(fn)
+
+	ret, err := CallNoPanic(madeFn, []interface{}{"tester"})
+	if err != nil {
+		t.Fatalf("CallNoPanic(madeFn, [\"tester\"]) errored %v, want nil", err)
```
[the remark this answers is not in the document]
Comment (damccorm) [reply sdks/go/pkg/beam/core/util/reflectx/call_test.go:71]: Yeah that makes sense, good call - updated!

**Pull request merged**